Source/ACE/Entity/Lifestone.cs

```diff
@@ -1,63 +1,88 @@
-                                                                 player.Sequences, motionSanctuary);
-
-                // This event was present for a pcap in the training dungeon.. Why? The sound comes with animationEvent...
-                var soundEvent = new GameMessageSound(Guid, Sound.LifestoneOn, 1);
-
-                if (player.Location.SquaredDistanceTo(Location) >= radiusSquared)
+                if (player == null)
                 {
-                    serverMessage = "You wandered too far to attune with the Lifestone!";
+                    return;
                 }
+
+                if (!player.IsWithinUseRadiusOf(this))
+                    player.DoMoveTo(this);
                 else
                 {
-                    player.SetCharacterPosition(PositionType.Sanctuary, player.Location);
+                    ActionChain useObjectChain = new ActionChain();
 
-                    // create the outbound server message
-                    serverMessage = "You have attuned your spirit to this Lifestone. You will resurrect here after you die.";
-                    player.HandleActionMotion(motionSanctuary);
-                    player.Session.Network.EnqueueSend(soundEvent);
-                }
+                    useObjectChain.AddAction(player, () =>
+                    {
+                        Activate(player);
+                    });
 
-                var lifestoneBindMessage = new GameMessageSystemChat(serverMessage, ChatMessageType.Magic);
-                // always send useDone event
-                var sendUseDoneEvent = new GameEventUseDone(player.Session);
-                player.Session.Network.EnqueueSend(lifestoneBindMessage, sendUseDoneEvent);
+                    useObjectChain.AddDelaySeconds(5); // TODO: get animation frames length and put that delay here
+
+                    useObjectChain.AddAction(player, () => player.SetCharacterPosition(PositionType.Sanctuary, player.Location));
+
+                    useObjectChain.AddAction(player, () => player.Session.Network.EnqueueSend(new GameMessageSystemChat("You have attuned your spirit to this Lifestone. You will resurrect here after you die.", ChatMessageType.Magic)));
+
+                    var sendUseDoneEvent = new GameEventUseDone(player.Session);
+                    useObjectChain.AddAction(player, () => player.Session.Network.EnqueueSend(sendUseDoneEvent));
+
+                    useObjectChain.EnqueueChain();
+                }
             });
             chain.EnqueueChain();
         }
+
+        private void Activate(Player activator)
+        {
+            CurrentLandblock.EnqueueBroadcastMotion(activator, motionSanctuary);
+
+            CurrentLandblock.EnqueueBroadcastSound(this, Sound.LifestoneOn); // This event was present for a pcap in the training dungeon.. Why? The sound already comes with animationEvent...
+
+            if (activator.Guid.Full > 0)
+                UseTimestamp++;
```

Comment:
actually, it is in fact. Clearly it can be done at least two different ways. It can only be considered misleading if one simply looks at the line itself and never looks at the definition nor the implementation. 

The silly thing about this PR is its mostly reused code already in the project, yet it's picked apart like its brand new never before seen code. 24 discussion points on something so simple is just absurd.

A lot of "Perfect is the enemy of good" situations feels like these days.